Mlem/App/Views/Shared/PostGridView.swift

```diff
@@ -38,10 +63,23 @@ struct PostGridView: View {
                             GridItem(.flexible(), spacing: 0, alignment: .leading)
                         ]
                     }
-                } else {
-                    // Avoid causing unnecessary view update
-                    if columns.count == 2 {
-                        columns = [GridItem(.flexible())]
+                } else if columns.count > 1 {
+                    // Only trigger if not already 1 column to avoid causing unnecessary view update
+                    columns = [GridItem(.flexible())]
+                }
+            }
+            .toolbar {
+                ToolbarEllipsisMenu {
+                    if let actions {
+                        ForEach(actions, id: \.id) { action in
+                            MenuButton(action: action)
+                        }
+                    }
+                    
+                    Divider()
+                    
+                    ForEach(standardActions, id: \.id) { action in
+                        MenuButton(action: action)
                     }
                 }
             }
```

Comment:
The community actions aren't shown if you switch tabs to "About" in `CommunityView`. This could be fixed by duplicating the toolbar actions in `CommunityView` to cover those cases.

Or alternatively, using `ToolbarItemGroup(placement: .secondaryAction)` lets you put groups of actions from multiple different views in the same ellipsis menu. There could be a `ToolbarItemGroup` in `PostGridView` for the feed actions, and a `ToolbarItemGroup` in `CommunityView` for the community actions. That also avoids having to pass `[any Action]` to `PostGridView`. 

iirc `Divider` doesn't work when using the above system, though the same effect can be achieved using `Section`. The downside of using that system is that the order in which the groups are rendered can't be easily controlled (afaik). I set this up one time in v1 and it was in the right order for our use case, so it should be possible.

Reply:
Nice catch! The `ToolbarItemGroup` seems like the most canonical way to handle it--it sets them up correctly in this case, and if we run into issues down the line we can revisit and develop a more sophisticated solution.